slsDetectorSoftware/src/Module.cpp

```diff
@@ -3600,6 +3604,29 @@ sls_detector_module Module::readSettingsFile(const std::string &fname, int tb) {
         LOG(logDEBUG1) << "tau:" << myMod.tau;
     }
 
+    // mythen3 (dacs, trimbits)
```

Comment:
I think it would be nice to have specialized functions, or a function that takes detector type as a parameter. 
These functions could be outside of the Module class then we could thest them easily. 

More for the eiger function, you could also return early if the file is not good. Meaning that you check !infile.good() instead of having  the deep nesting.